pkg/query-service/auth/auth.go

```diff
@@ -276,17 +334,40 @@ func Register(ctx context.Context, req *RegisterRequest) *model.ApiError {
 		CreatedAt:          time.Now().Unix(),
 		ProfilePirctureURL: "", // Currently unused
 		GroupId:            group.Id,
-		OrgId:              orgId,
+		OrgId:              invite.OrgId,
 	}
 
 	// TODO(Ahsan): Ideally create user and delete invitation should happen in a txn.
-	_, apiErr = dao.DB().CreateUser(ctx, user)
+	user, apiErr = dao.DB().CreateUser(ctx, user)
 	if apiErr != nil {
 		zap.S().Debugf("CreateUser failed, err: %v\n", apiErr.Err)
-		return apiErr
+		return nil, apiErr
 	}
 
-	return dao.DB().DeleteInvitation(ctx, user.Email)
+	apiErr = dao.DB().DeleteInvitation(ctx, user.Email)
+	if apiErr != nil {
+		zap.S().Debugf("delete invitation failed, err: %v\n", apiErr.Err)
+		return nil, apiErr
+	}
+
+	return user, nil
+}
+
+// Register registers a new user. For the first register request, it doesn't need an invite token
+// and also the first registration is an enforced ADMIN registration. Every subsequent request will
+// need an invite token to go through.
+func Register(ctx context.Context, req *RegisterRequest) (*model.User, *model.ApiError) {
+	users, err := dao.DB().GetUsers(ctx)
+	if err != nil {
+		return nil, model.InternalError(fmt.Errorf("failed to get user count"))
+	}
+
+	switch len(users) {
```

Comment:
This is not a frequently used request path but could be changed to use a method that just returns count or bool.

Reply:
Actually I refactored this code and don't really think it should be this way. But didn't want to introduce major change in approach. 

We should have a better way to knowing if this is fresh installation, especially in case of cloud (SaaS) scenario this code would be less than ideal. 

Perhaps, we can log an issue and handle this in a separate PR.